src/SFML/Graphics/Text.cpp

```diff
@@ -540,4 +577,74 @@ void Text::ensureGeometryUpdate() const
     m_bounds.height = maxY - minY;
 }
 
+
+////////////////////////////////////////////////////////////
+void Text::updateLineOffsets() const
+{
+    // widths of each line
+    std::vector<float> lineWidths;
+
+    // Precompute the variables needed by the algorithm
+    const bool  isBold          = m_style & Bold;
+    float       whitespaceWidth = m_font->getGlyph(U' ', m_characterSize, isBold).advance;
+    const float letterSpacing   = (whitespaceWidth / 3.f) * (m_letterSpacingFactor - 1.f);
+    whitespaceWidth += letterSpacing;
+    const float lineSpacing = m_font->getLineSpacing(m_characterSize) * m_lineSpacingFactor;
+
+    Vector2f      position;
+    std::uint32_t prevChar = 0;
+    float         maxWidth = 0.f;
+    for (std::size_t i = 0; i < m_string.getSize(); ++i)
+    {
+        const std::uint32_t curChar = m_string[i];
+
+        // Apply the kerning offset
+        position.x += m_font->getKerning(prevChar, curChar, m_characterSize, isBold);
+        prevChar = curChar;
+
+        // Handle special characters
+        switch (curChar)
+        {
+            case U' ':
+                position.x += whitespaceWidth;
+                continue;
+            case U'\t':
+                position.x += whitespaceWidth * 4;
+                continue;
+            case U'\n':
+                position.y += lineSpacing;
+                if (position.x > maxWidth)
+                    maxWidth = position.x;
+                lineWidths.push_back(position.x);
+                position.x = 0;
+                continue;
+        }
+
+        // For regular characters, add the advance offset of the glyph
+        position.x += m_font->getGlyph(curChar, m_characterSize, isBold).advance + letterSpacing;
+    }
+
+    // add final part of the string since last newline as the final line
+    // (this is the entire string if text is single line without newlines)
+    lineWidths.push_back(position.x);
+
+    // update line offsets from widths depending on line alignment
+    m_lineOffsets.resize(lineWidths.size());
```

Comment:
Could we use 1 vector here instead of 2? As in, could we remove `lineWidths` and replace that with a more clever use of `m_ lineWidths`?

Reply:
Yes. It was briefly mentioned early here but wasn't sure if it would be clear to use the same vector for two semantically different things: widths and offsets, even if it is temporary, private and "internal".